### PR TITLE
feat: add feature nat module

### PR DESCRIPTION
**Pull request by llewita**

### CODE REVIEW

File: .terraform-docs.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+recursive:
+  enabled: false
+  path: ""
+  
+sections:
+  hide: []
+  show: []  
+  
+  hide-all: false # deprecated in v0.13.0, removed in v0.15.0
+  show-all: true  # deprecated in v0.13.0, removed in v0.15.0
```
Comment (azrod) [inline .terraform-docs.yml:13]: Actual version of terraform-docs is v0.16.0

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+resource "cloudavenue_edgegateway_nat_rule" "example" {
```
Comment (azrod) [inline main.tf:1]:
```suggestion
resource "cloudavenue_edgegateway_nat_rule" "nat-rule" {
```

File: tests/setup/main.tf [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+provider "cloudavenue" {
+  org      = "my_org"
+  user     = "my_user"
+  password = "my_password"
+}
+
+data "cloudavenue_tier0_vrfs" "example" {}
+
+resource "cloudavenue_vdc" "example" {
+  name                  = "ModuleTF"
```
Comment (azrod) [inline tests/setup/main.tf:19]: Randomize name

File: tests/setup/main.tf [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+module "nat_rules" {
+  source = "./../.."
+  rules = [
+    {
+      "name"                      = "example-dnat"
+      "edge_gateway_name"         = cloudavenue_edgegateway.example.name
+      "rule_type"                 = "DNAT"
+      "description"               = "description DNAT example Updated!!"
```
Comment (azrod) [inline tests/setup/main.tf:51]:
```suggestion
      "description"               = "description DNAT example"
```

**Review by azrod — CHANGES_REQUESTED** [review]
Add result of unit test

**Comment by llewita** [comment]
Example of success running test:
```
terraform test
tests\nat.tftest.hcl... in progress
  run "setup_tests"... pass
tests\nat.tftest.hcl... tearing down
tests\nat.tftest.hcl... pass

Success! 1 passed, 0 failed.
```

**Review by azrod — APPROVED** [review]

**Pull request merged**